DigitalLearningSolutions.Web/Services/UserService.cs

```diff
@@ -393,25 +416,55 @@ bool changeMadeBySameUser
                     delegateDetailsData.Answer4,
                     delegateDetailsData.Answer5,
                     delegateDetailsData.Answer6,
-                    detailsLastChecked
+                    currentTime
                 );
             }
 
-            UpdateUserDetails(editAccountDetailsData, changeMadeBySameUser, detailsLastChecked);
+            UpdateUserDetails(editAccountDetailsData, isPrimaryEmailUpdated, changeMadeBySameUser, currentTime);
 
-            userDataService.SetCentreEmail(
-                editAccountDetailsData.UserId,
-                centreId,
-                centreEmail,
-                changeMadeBySameUser ? (DateTime?)null : clockUtility.UtcNow
-            );
+            if (isCentreEmailUpdated)
+            {
+                var emailVerified =
+                    changeMadeBySameUser &&
+                    !emailVerificationDataService
+                        .AccountEmailIsVerifiedForUser(
+                            editAccountDetailsData.UserId,
+                            centreEmail
+                        )
+                        ? (DateTime?)null
+                        : currentTime;
+
+                userDataService.SetCentreEmail(
+                    editAccountDetailsData.UserId,
+                    centreId,
+                    centreEmail,
+                    emailVerified
+                );
+            }
         }
 
-        public void SetCentreEmails(int userId, Dictionary<int, string?> centreEmailsByCentreId)
+        public void SetCentreEmails(
+            int userId,
+            Dictionary<int, string?> centreEmailsByCentreId,
+            List<UserCentreDetails> userCentreDetails
+        )
         {
+            var currentTime = clockUtility.UtcNow;
+
             foreach (var (centreId, email) in centreEmailsByCentreId)
             {
-                userDataService.SetCentreEmail(userId, centreId, email);
+                if (!string.Equals(
+                        email,
+                        userCentreDetails.Where(ucd => ucd.CentreId == centreId).Select(ucd => ucd.Email)

```

Comment:
I think this can be `userCentreDetails.SingleOrDefault(ucd => ucd.CentreId == centreId)?.Email`